src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,88 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int SHIFT_BY_ONE = 1;
+    private Node<K, V>[] table;
+    private int size;
+    private int maxCapacity;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        maxCapacity = (int) (table.length * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (maxCapacity < size) {
+            resize();
+        }
+        int index = hash(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+        } else {
+            Node<K, V> node = table[index];
+            while (node != null) {
+                if (node.key == key || node.key != null && node.key.equals(key)) {
```

Comment:
It looks like here and on the 46 line you have code duplication. Consider moving it to the separate method